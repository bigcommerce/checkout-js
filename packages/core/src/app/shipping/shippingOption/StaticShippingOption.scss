@import '../../ui/Base';

// Static Shipping Option
// ---------------------------------------------------
.shippingOption {
    display: inline-block;
    min-height: 0;
    width: 100%;
}

.shippingOption--selected {
    border-top: container("border");
    margin-top: spacing("half");
    padding-top: spacing("half");
}

.shippingOption-figure,
.shippingOption-desc,
.shippingOption-price {
    display: inline-block;
    font-size: fontSize("base");
}

.shippingOption-img {
    padding-right: spacing("half");
}

.shippingOption-price {
    font-weight: fontWeight("bold");
    padding-left: spacing("half");
}

.shippingOption-transitTime {
    display: block;
    font-size: fontSize("base");
    font-weight: fontWeight("normal");
}

// ShippingOption Label
// ---------------------------------------------------
.shippingOptionLabel {
    .shippingOption {
        align-items: center;
        display: flex;
        min-height: remCalc(35px);
    }

    .shippingOption-figure,
    .shippingOption-desc,
    .shippingOption-price {
        font-size: inherit;
        line-height: lineHeight("base");
        vertical-align: middle;
    }

    .shippingOption-figure,
    .shippingOption-desc {
        flex: 1;
    }

    .shippingOption-figure {
        max-width: remCalc(50px);
        min-width: remCalc(50px);
    }

    .shippingOption-additionalDescription--container {
        display: flex;
        flex-direction: row;
        line-height: 1.5rem;

        .shippingOption-additionalDescription {
            font-size: fontSize("smaller");
            font-weight: fontWeight("normal");
            text-overflow: ellipsis;
        }

        .shippingOption-additionalDescription--collapsed {
            height: 1.5rem;
            overflow: hidden;
            width: 65%;
        }

        .shippingOption-additionalDescription--expanded {
            height: auto;
            overflow: visible;
            white-space: initial;
            width: 100%;
        }

        .shippingOption-readMore {
            font-weight: normal;
        }
    }
}
