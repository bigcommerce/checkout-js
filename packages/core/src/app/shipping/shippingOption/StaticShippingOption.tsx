import { ShippingOption } from '@bigcommerce/checkout-sdk';
import React from 'react';

import { ShopperCurrency } from '../../currency';

import ShippingOptionAdditionalDescription from './ShippingOptionAdditionalDescription';
import './StaticShippingOption.scss';

interface StaticShippingOptionProps {
    displayAdditionalInformation?: boolean;
    method: ShippingOption;
}

const StaticShippingOption: React.FunctionComponent<StaticShippingOptionProps> = ({
displayAdditionalInformation = true,
method,
}) => {
    return (
        <>
            <div className="shippingOption shippingOption--alt">
                { method.imageUrl &&
                    <span className="shippingOption-figure">
                        <img
                            alt={ method.description }
                            className="shippingOption-img"
                            src={ method.imageUrl }
                        />
                    </span> }
                <span className="shippingOption-desc">
                    { method.description }
                    { method.transitTime &&
                        <span className="shippingOption-transitTime">
                            { method.transitTime }
                        </span> }
                    { method.additionalDescription && displayAdditionalInformation &&
                        <ShippingOptionAdditionalDescription description={ method.additionalDescription } /> }
                </span>
                <span className="shippingOption-price">
                    <ShopperCurrency amount={ method.cost } />
                </span>
            </div>
        </>
    );
};

export default StaticShippingOption;
