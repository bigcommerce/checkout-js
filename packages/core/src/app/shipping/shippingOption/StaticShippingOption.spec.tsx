import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import { getShippingOption } from './shippingMethod.mock';
import StaticShippingOption from './StaticShippingOption';

describe('StaticShippingOption Component', () => {
    const method = getShippingOption();

    it('renders static shipping option with optional information', () => {
        const tree = shallow(<StaticShippingOption method={method} />);

        expect(toJson(tree)).toMatchSnapshot();
    });

    it('renders static shipping option with minimum information', () => {
        const tree = shallow(
            <StaticShippingOption
                method={{
                    ...method,
                    imageUrl: '',
                    transitTime: '',
                }}
            />,
        );

        expect(toJson(tree)).toMatchSnapshot();
    });
});
