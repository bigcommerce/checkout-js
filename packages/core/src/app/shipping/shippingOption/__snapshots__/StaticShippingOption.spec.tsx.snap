// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StaticShippingOption Component renders static shipping option with minimum information 1`] = `
<div
  className="shippingOption shippingOption--alt"
>
  <span
    className="shippingOption-desc"
  >
    Flat Rate
    <Memo(ShippingOptionAdditionalDescription)
      description="Pick up in store additional description"
    />
  </span>
  <span
    className="shippingOption-price"
  >
    <WithCurrency(ShopperCurrency)
      amount={0}
    />
  </span>
</div>
`;

exports[`StaticShippingOption Component renders static shipping option with optional information 1`] = `
<div
  className="shippingOption shippingOption--alt"
>
  <span
    className="shippingOption-desc"
  >
    Flat Rate
    <Memo(ShippingOptionAdditionalDescription)
      description="Pick up in store additional description"
    />
  </span>
  <span
    className="shippingOption-price"
  >
    <WithCurrency(ShopperCurrency)
      amount={0}
    />
  </span>
</div>
`;
